vector/src/main/java/im/vector/app/core/platform/VectorBaseDialogFragment.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.app.core.platform
+
+import android.content.Context
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.CallSuper
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import androidx.viewbinding.ViewBinding
+import com.airbnb.mvrx.MavericksView
+import dagger.hilt.android.EntryPointAccessors
+import im.vector.app.R
+import im.vector.app.core.di.ActivityEntryPoint
+import im.vector.app.core.extensions.singletonEntryPoint
+import im.vector.app.core.extensions.toMvRxBundle
+import im.vector.app.features.analytics.AnalyticsTracker
+import im.vector.app.features.analytics.plan.MobileScreen
+import im.vector.app.features.themes.ThemeUtils
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import reactivecircus.flowbinding.android.view.clicks
+import timber.log.Timber
+
+/**
+ * Add Mavericks capabilities, handle DI and bindings.
+ */
+abstract class VectorBaseDialogFragment<VB : ViewBinding> : DialogFragment(), MavericksView {
```

Comment:
Why creating this new baseclass ? the ui doesn't look like a Dialog. Also we use BottomSheet usually? It's weird

Reply:
The reason I didn't use a bottom sheet because the design is supposed to look like a 'full screen dialog' (see this in the [Material spec](https://m3.material.io/components/dialogs/specs#bbf1acde-f8d2-4ae1-9d51-343e96c4ac20)). I didn't want to fight against the bottom sheet dialog implementation to achieve this full screen design.

Similar features' UI (e.g. polls) use a new Activity to achieve this design but I didn't want to introduce a whole new Activity for this simple modal interaction. I also wanted to be able to communicate to the calling Fragment using a ViewModel which is not possible from a separate Activity.

Hope that makes sense?

Reply:
Got it, lets keep it like this then, thanks!